internals/daemon/daemon.go

```diff
@@ -531,22 +529,20 @@ func (d *Daemon) Start() error {
 func (d *Daemon) HandleRestart(t restart.RestartType) {
```

Comment:
‼️ this can be called from multiple goroutines for each service during shutdown. We need to be careful here. ‼️

Reply:
Since HandleRestart is called from (serviceData).exited (which is called from the goroutine from (serviceData).startInternal), it can be called from multiple dying services which might have different actions on exit. This could result in some strange behaviour during exit.